### PR TITLE
Refactor file dialog service to fix layer breaker

### LINKED ISSUE #81161: File dialog service breaks layers
While trying to move methods from `IWindowService` to `IElectronService` I noticed that `fileDialogService` is a heavy consumer of the native Electron dialogs. 

I see a couple of these calls:

```ts
const shouldUseSimplified = this.shouldUseSimplified(schema);
```

In my mind, we need to separate a `browser` `IFileDialogService` implementation from the `electron-browser` one. All the usages of Electron API then can only live in the `electron-browser` implementation. I think we already do something similar for the dialog service.

I want to get this done for this milestone, so assigning to September. Let me know if you need any help. 

For now I am pushing a change where I have an optional `IElectronService` in the file dialog service which of course is a clear layer breaker and I added a "fix" for the warning. 

Refs: https://github.com/microsoft/vscode/commit/af00f97482e4d8fd9755f12eb247732502a17155, https://github.com/microsoft/vscode/commit/8fb28a37e9c52d3c0a43659615d4a78a4428655f

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #81161

### CODE REVIEW

**Comment by alexr00** [comment]
@aeschli I think you are the person who owns the file dialog service the most. This is a very simple refactor (only moving code, not other changes) to fix a layer breaker exposed by Ben's electron service changes. Do you want to review this before I merge?

**Review by bpasero — APPROVED** [review]

File: src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+		// Include File schema unless the schema is web
+		// Don't allow untitled schema through.
+		if (isWeb) {
+			return schema === Schemas.untitled ? [Schemas.file] : [schema];
+		} else {
+			return schema === Schemas.untitled ? [Schemas.file] : (schema !== Schemas.file ? [schema, Schemas.file] : [schema]);
+		}
+	}
+
+	protected async pickFileFolderAndOpenSimplified(schema: string, options: IPickAndOpenOptions, shouldUseSimplifiedSetting: boolean): Promise<any> {
```
Comment (aeschli) [inline src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts]: rename to 'preferCurrentWindow'

File: src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+	protected toNativeOpenDialogOptions(options: IPickAndOpenOptions): INativeOpenDialogOptions {
+		return {
+			forceNewWindow: options.forceNewWindow,
+			telemetryExtraData: options.telemetryExtraData,
+			defaultPath: options.defaultUri && options.defaultUri.fsPath
+		};
+	}
+
+	protected shouldUseSimplified(schema: string): { useSimplified: boolean, isSetting: boolean } {
```
Comment (aeschli) [inline src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts]: Needs to go up to Electron

**Review by aeschli — CHANGES_REQUESTED** [review]

File: src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ import { URI } from 'vs/base/common/uri';
 import { Schemas } from 'vs/base/common/network';
 import * as resources from 'vs/base/common/resources';
 import { IInstantiationService, } from 'vs/platform/instantiation/common/instantiation';
-import { RemoteFileDialog } from 'vs/workbench/services/dialogs/browser/remoteFileDialog';
+import { SimpleFileDialog as SimpleFileDialog } from 'vs/workbench/services/dialogs/browser/simpleFileDialog';
```
Comment (aeschli) [inline src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts]: unnecessary code generated by ts-server...

File: src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts [hunk 4]
```diff
@@ -102,31 +88,31 @@ export class AbstractFileDialogService {
...
 
-		const uri = await this.pickRemoteResource({ canSelectFiles: true, canSelectFolders: true, canSelectMany: false, defaultUri: options.defaultUri, title, availableFileSystems });
+		const uri = await this.pickResource({ canSelectFiles: true, canSelectFolders: true, canSelectMany: false, defaultUri: options.defaultUri, title, availableFileSystems });
 
 		if (uri) {
 			const stat = await this.fileService.resolve(uri);
 
 			const toOpen: IURIToOpen = stat.isDirectory ? { folderUri: uri } : { fileUri: uri };
-			if (stat.isDirectory || options.forceNewWindow || shouldUseSimplifiedSetting) {
+			if (stat.isDirectory || options.forceNewWindow || preferCurrentWindow) {
```
Comment (aeschli) [inline src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts]: do we need to rename the option? looks like it's the opposite,

**Review by aeschli — APPROVED** [review]

**Pull request merged**